src/gpio.rs

```diff
@@ -1373,6 +1376,159 @@ macro_rules! pin {
         impl_touch!($pxi: $pin, $touch: $touchno);
     };
 }
+#[cfg(feature = "nightly")]
```

Comment:
No need for `nightly` as none of the APIs here use a nightly feature. If you feel that the async APIs are kind of "not quite ready yet" we can hide them behind the `experimental` feature instead, which is orthogonal to `nightly` (i.e. something can be still marked as experimental, even if it does not use nightly features).

Also I think you need `riscv-ulp-hal` here.